roles/mysql/handlers/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+# handlers file for mysql
+#
+- name: systemctl daemon-reexec
+  ansible.builtin.systemd:
+    daemon_reexec: true
+  when:
+    - running_on_server
+
+- name: restart mysql server
+  ansible.builtin.service:
+    name: mysql
+    state: restarted
+  when:
+    - not ansible_check_mode | bool
```

Comment:
I don't think we need to state the data type here. `check_mode` is a boolean magic var, set to no/false by default. It gets set to yes/true when you run a play with `- check`. See https://docs.ansible.com/ansible/latest/user_guide/playbooks_checkmode.html#skipping-tasks-or-ignoring-errors-in-check-mode.